Demos/libraries/setup.py

```diff
@@ -2,30 +2,47 @@
 
 import os
 import sys
+import sysconfig
 
 from distutils.core import setup
 from distutils.extension import Extension
 from Cython.Build import cythonize
 
 
 # For demo purposes, we build our own tiny library.
-try:
-    print("building libmymath.a")
-    assert os.system("gcc -shared -fPIC -c mymath.c -o mymath.o") == 0
-    assert os.system("ar rcs libmymath.a mymath.o") == 0
-except:
-    if not os.path.exists("libmymath.a"):
-        print("Error building external library, please create libmymath.a manually.")
-        sys.exit(1)
+if not sys.platform.startswith("win"):
+    try:
+        print("building libmymath.a")
+        assert os.system(f"gcc -I{sysconfig.get_path("include")} -shared -fPIC -c mymath.c -o mymath.o") == 0
+        assert os.system("ar rcs libmymath.a mymath.o") == 0
+    except Exception as ex:
+        print(ex)
+        if not os.path.exists("libmymath.a"):
+            print("Error building external library, please create libmymath.a manually.")
+            sys.exit(1)
```

Comment:
It looks like this part can be pulled out of the build cases and done generally, in the end. (I.e. do "try … if … elif" instead of "if … try … elif … try".)

Reply:
Variables applied to handle platform specific library name.
Added *.split(os.path.pathsep) to make directory list parameter portable
Hopefully this closes the gap